### PR TITLE
build-script: build static version of libdispatch

### PR DESCRIPTION
**Pull request by compnerd**
build and install libdispatch static.  We now build libdispatch using
CMake, which does not easily generate static and shared versions of
libraries.  Create two builds to populate the toolchain distribution.

Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves [SR-NNNN](https://bugs.swift.org/browse/SR-NNNN).

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Comment by compnerd** [comment]
@swift-ci please build toolchain

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Comment by compnerd** [comment]
@swift-ci please build toolchain

**Comment by compnerd** [comment]
Please test with following PRs:
https://github.com/apple/swift-corelibs-libdispatch/pull/433

@swift-ci please test

**Comment by compnerd** [comment]
Please test with following PRs:
https://github.com/apple/swift-corelibs-libdispatch/pull/433

@swift-ci please build toolchain

**Comment by compnerd** [comment]
Please test with following PRs:
apple/swift-corelibs-libdispatch#433

@swift-ci please build toolchain

**Comment by compnerd** [comment]
Please test with following PRs:
apple/swift-corelibs-libdispatch#433

@swift-ci please build toolchain

**Comment by compnerd** [comment]
@swift-ci please smoke test and merge

**Comment by compnerd** [comment]
@swift-ci please smoke test and merge

File: utils/build-script-impl [hunk 1]
```diff
@@ -3282,6 +3286,11 @@ for host in "${ALL_HOSTS[@]}"; do
                 ;;
                 esac
                 ;;
+            libdispatch_static)
+              # FIXME: merge with libdispatch once the unit tests work with
```
Comment (kevints) [inline utils/build-script-impl]: Is there a bug for this follow-up work?

Comment (compnerd) [reply utils/build-script-impl]: No, I didn't create a SR for that.

**Pull request merged**